Serialize values with returned type

amy-hs opened the pull request:
Passing a column to the method `type_cast` is deprecated  and will be removed in Rails 7. For now, we can manually call `lookup_cast_type_from_column` and serialize our values with the returned type.

amy-hs committed:
Serialize values with returned type

amy-hs committed:
Try using an older version of bundler

timlkelly committed:
Use ruby/setup-ruby@v1

timlkelly committed:
Remove version comment

timlkelly commented on .github/workflows/ruby.yml:
```diff
@@ -46,11 +46,10 @@ jobs:
           if [ "$DB" = "mysql" ]; then sudo systemctl start mysql.service; fi
           if [ "$DB" = "mysql" ]; then mysql -e 'create database IF NOT EXISTS fast_inserter;' -uroot -proot; fi
       - name: Set up Ruby
-        # Pin to this commit: v1.82.0
```
@roland-at-handshake, any reason to pin to this version? the workflow wasn't able to run successfully but using the latest fixes it.

roland-at-handshake replied:
I pinned it to give more control over the action that was used. For example, we wouldn't auto-upgrade to a version that introduced a new vulnerability, or was somehow backwards incompatible (thereby breaking us without any changes on our part). I don't feel too strongly about it, but that was my rationale.

timlkelly committed:
Update to account for MySQL changes

ActiveRecord improved the security of untyped bound values in version 7.
This causes ActiveRecord to return values as Strings. See the pull for details
https://github.com/rails/rails/pull/42440

timlkelly committed:
Add additional ruby versions to test

timlkelly committed:
Fix gh workflow quotes

timlkelly committed:
Remove Rake's version limitation

timlkelly committed:
Remove EoL Ruby versions 2.4 and 2.5

timlkelly committed:
Use 2.6.10 and 2.7.8 rubies

timlkelly commented:
Required us to update a spec due to the behavior changes in the ActiveRecord MySQL adapter described [here](https://github.com/rails/rails/pull/42440/files).

Additionally, I removed 2.4 and 2.5 from testing. These versions are well past their end-of-life date.

timlkelly commented:
@amy-hs I can't tag you to review because this is already your pull request.

amy-hs commented:
> @amy-hs I can't tag you to review because this is already your pull request.

I think we'll need to mutually review 😅 I tagged @roland-at-handshake as well.

amy-hs commented on spec/spec_helper.rb:
```diff
@@ -30,3 +30,7 @@
+
+def mysql_security_type_casting?
+  ENV['DB'] == 'mysql' && Gem::Version.new(ActiveRecord.version) >= Gem::Version.new('7.0.0.0')
+end
```
🎸

roland-at-handshake commented on .github/workflows/ruby.yml:
```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       matrix:
-        ruby-version: ['2.4.0', '2.5.1', '2.6.6', '2.7.2', 'ruby-head']
+        ruby-version: ['2.6.10', '2.7.8', '3.0.6', '3.1.4', '3.2.2', 'ruby-head']
```
By removing these versions from the matrix, we are effectively dropping support for them. 

My thinking is this means we need to bump versions and update the changelog. I'm not well-versed enough with semver etiquette to say whether this would be a minor or a major version bump. There are no breaking functional changes, but we are dropping explicit support, so future breaking changes could be added, and we wouldn't catch them in the test suite.

amy-hs replied:
Yup! We plan to release a major version bump, so folks can still use the old one if they want.

timlkelly replied:
@roland-at-handshake @amy-hs Updated the changelog and version. Once this is merged, I will run the release script to push to rubygems. 

I try to follow this document on how to use semver and when to bump what, https://semver.org/. I believe this warrants a major bump but open to other thoughts.

timlkelly committed:
Update .ruby-version

timlkelly committed:
Update CHANGELOG and bump version
